src/waypoint.jsx

```diff
@@ -333,13 +352,30 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.noWrapper) {
+      const child = React.Children.only(this.props.children);
+      const ref = (node) => {
+        this.refElement(node);
+        if (this.props.children.ref) {
+          this.props.children.ref(node);
+        }
+      };
+      return React.cloneElement(child, { ref });
+    }
+
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
+    }
+
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
     return <span ref={this.refElement} style={{ fontSize: 0 }} />;
   }
 }
 
 Waypoint.propTypes = {
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
+  noWrapper: PropTypes.bool,
```

Comment:
> That's unnecessary clutter of user's code.

I disagree about this. It makes it very clear what is going on and puts full control in the hands of the developer. Of course, consumers could choose to abstract this away in a component that wraps Waypoint and adds a div, or we could provide a separate component that does this for folks.

```jsx
export default function WaypointDiv({ children, ...waypointProps }) {
  return <Waypoint {...waypointProps}><div>{children}</div></Waypoint>;
}
```